src/integer_mod_q/modulus_polynomial_ring_zq/from.rs

```diff
@@ -89,15 +234,156 @@ impl FromStr for ModulusPolynomialRingZq {
+///
+/// Returns an empty `Ok`, or an [`MathError`] if the polynomial is no valid modulus polynomial.
+///
+/// # Examples
+/// ```compile_fail
+/// use qfall_math::integer_mod_q::PolyOverZq;
+/// use std::str::FromStr;
+///
+/// let poly_zq = PolyOverZq::from_str("2  1 2 mod 17").unwrap();
+///
+/// check_poly_mod(&poly_zq)?
+/// ```
+///
+/// # Errors and Failures
+/// - Returns a [`MathError`] of type
+///     [`InvalidModulus`](MathError::InvalidModulus)
+///     if the modulus polynomial is `0` or `1`.
+pub(crate) fn check_poly_mod(poly_zq: &PolyOverZq) -> Result<(), MathError> {
+    if poly_zq == &PolyOverZq::from((1, &poly_zq.modulus))
+        || poly_zq == &PolyOverZq::from((0, &poly_zq.modulus))
+    {
+        return Err(MathError::InvalidModulus(poly_zq.to_string()));
+    }
+
+    Ok(())
+}
+
+/// Most tests with specific values are covered in [`PolyOverZq`](crate::integer_mod_q::PolyOverZq)
+/// since the format is reused, we omit some tests
+#[cfg(test)]
+mod test_availability {
+    use super::*;
+    use crate::{integer::Z, integer_mod_q::Zq};
+    use crate::{integer_mod_q::ModulusPolynomialRingZq, integer_mod_q::PolyOverZq};
+    use std::str::FromStr;
+
+    /// Ensure that the from function can be called with several types.
+    #[test]
+    fn availability() {
+        let z = Z::from(4);
+        let modulus = Modulus::from(3);
+        let zq = Zq::from((2, 3));
+        let poly = PolyOverZ::from_str("2  1 1").unwrap();
+        let poly_zq = PolyOverZq::from_str("4  1 0 0 1 mod 17").unwrap();
+
+        let _ = ModulusPolynomialRingZq::from(&zq);
+        let _ = ModulusPolynomialRingZq::from(zq.clone());
+
+        let _ = ModulusPolynomialRingZq::from((2, 5));
+        let _ = ModulusPolynomialRingZq::from((&z, 5));
+        let _ = ModulusPolynomialRingZq::from((z.clone(), 5));
+        let _ = ModulusPolynomialRingZq::from((&modulus, 5));
+        let _ = ModulusPolynomialRingZq::from((modulus.clone(), 5));
+        let _ = ModulusPolynomialRingZq::from((&poly, 5));
+        let _ = ModulusPolynomialRingZq::from((poly.clone(), 5));
+        let _ = ModulusPolynomialRingZq::from((3, &z));
+        let _ = ModulusPolynomialRingZq::from((3, z.clone()));
+        let _ = ModulusPolynomialRingZq::from((2, &modulus));
+        let _ = ModulusPolynomialRingZq::from((2, modulus));
+
+        let _ = ModulusPolynomialRingZq::from(&poly_zq);
+        let _ = ModulusPolynomialRingZq::from(poly_zq);
+    }
+}
+
+/// Most tests with specific values are covered in [`PolyOverZq`](crate::integer_mod_q::PolyOverZq)
+/// since the format is reused, we omit some tests
+#[cfg(test)]
+mod test_try_from_poly_z {
+    use crate::{integer::PolyOverZ, integer_mod_q::ModulusPolynomialRingZq};
+    use std::str::FromStr;
+
+    /// Ensure that non-primes work
+    #[test]
+    fn poly_z_non_prime() {
+        let poly_z = PolyOverZ::from_str("2  2 2").unwrap();
+
+        let _ = ModulusPolynomialRingZq::from((poly_z, 10));
+    }
```

Comment:
If you add a test for a non-prime, why isn't there one for primes?
I don't think that you need to introduce two different tests. I believe that these tests should be renamed and their description updated to be more general.